featurebyte/feast/utils/registry_construction.py

```diff
@@ -288,13 +291,50 @@ def create_feast_feature_services(
             feature_services.append(feature_service)
         return feature_services
 
+    @classmethod
+    def _create_feast_registry_proto(
+        cls,
+        project_name: Optional[str],
+        feast_data_sources: List[FeastDataSource],
+        primary_entity_ids_to_feast_entity: Dict[Tuple[PydanticObjectId, ...], FeastEntity],
+        feast_feature_views: List[FeastFeatureView],
+        feast_feature_services: List[FeastFeatureService],
+    ) -> RegistryProto:
+        project_name = project_name or "featurebyte_project"
+        with tempfile.NamedTemporaryFile() as temp_file:
+            repo_config = RepoConfig(
+                project=project_name,
+                provider="local",
+                registry=RegistryConfig(
+                    registry_type="file",
+                    path=temp_file.name,
+                    cache_ttl_seconds=0,
+                ),
+            )
+
+            feature_store = FeastFeatureStore(config=repo_config)
+            registry = feature_store.registry
+            for data_source in feast_data_sources:
+                registry.apply_data_source(data_source=data_source, project=project_name)
+            for entity in primary_entity_ids_to_feast_entity.values():
+                registry.apply_entity(entity=entity, project=project_name)
+            for feature_view in feast_feature_views:
+                registry.apply_feature_view(feature_view=feature_view, project=project_name)
+            for feature_service in feast_feature_services:
+                registry.apply_feature_service(
+                    feature_service=feature_service, project=project_name
+                )
+            registry_proto = registry.proto()
+            return cast(RegistryProto, registry_proto)
```

Comment:
Revise this part as feast feature store auto generates project metadata stores in the registry. Without these metadata, we can't list the assets through feast feature store interface.